subprojects/testing-base/src/test/groovy/org/gradle/api/internal/tasks/testing/FailFastTestListenerTestInternal.groovy

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.testing
+
+import org.gradle.api.internal.tasks.testing.results.TestListenerInternal
+import org.gradle.api.tasks.testing.TestOutputEvent
+import org.gradle.api.tasks.testing.TestResult
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+class FailFastTestListenerTestInternal extends Specification {
```

Comment:
❌  This test is using mocking so much that it essentially repeats the implementation line by line. Only mock interactions that you care about. Use Stubs (or the real classes in case of simple data objects) for everything else.

Reply:
I've reworked this class to only mock out the `TestExecuter` and `TestListenerInternal` and used a bunch of the `Simple` implementation of the other (previously mocked) interfaces.  Let me know if the changes aren't what you were looking for.